tests/test_modeling_common.py

```diff
@@ -245,15 +245,18 @@ def _create_and_check_torchscript(self, config, inputs_dict):
             inputs = self._prepare_for_class(inputs_dict, model_class)["input_ids"]  # Let's keep only input_ids
 
             try:
-                traced_gpt2 = torch.jit.trace(model, inputs)
+                if model.__class__.__name__ in ["T5Model", "T5ForConditionalGeneration"]:
```

Comment:
Quite hacky here, but I didn't see another way....

Reply:
Isn't it more general as `is_encoder_decoder`?

Reply:
Think `bart` can run without requiring the `decoder_input_ids` @sshleifer...but I guess it would be cleaner to call it `encoder_decoder` here...we will have to slightly change Bart then.

Reply:
If I change to `is_encoder_decoder=True`, and do the corresponding changes for Bart, Bart hits an assert: 

```
assert attn_output.size() == (bsz * self.num_heads, tgt_len, self.head_dim)
```

I guess `decoder_input_ids` should not be the same length as `input_ids` in Bart @sshleifer ?

Reply:
I'll push the changes -> let's see if we want to change Bart accordingly or revert to the T5 hack...

Reply:
I wrote that assert and should have written an error message, but I can't understand the issue without looking more closely. Does it matter that the second arg to `BartForConditionalGeneration.forward` is `attention_mask`?

Reply:
Its outdated, seems like you figured it out! I will add a message to my assert.